src/libraries/System.Private.CoreLib/src/System/Numerics/Vector_1.cs

```diff
@@ -0,0 +1,1507 @@
+
+    /// <summary>
+    /// A structure that represents a single Vector. The count of this Vector is fixed but CPU register dependent.
+    /// This struct only supports numerical types. This type is intended to be used as a building block for vectorizing
+    /// large algorithms. This type is immutable, individual elements cannot be modified.
+    /// </summary>
+    [Intrinsic]
+    public struct Vector<T> : IEquatable<Vector<T>>, IFormattable where T : struct
+    {
+        private ulong _00;
+        private ulong _01;
+
+        /// <summary>Returns the number of elements stored in the vector. This value is hardware dependent.</summary>
+        public static int Count
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return Unsafe.SizeOf<Vector<T>>() / Unsafe.SizeOf<T>();
+            }
+        }
+
+        /// <summary>Returns a vector containing all zeroes.</summary>
+        public static Vector<T> Zero
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return default;
+            }
+        }
+
+        /// <summary>Returns a vector containing all ones.</summary>
+        public static Vector<T> One
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return new Vector<T>(GetOneValue());
+            }
+        }
+
+        internal static Vector<T> AllBitsSet
+        {
+            [Intrinsic]
+            get
+            {
+                ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+                return new Vector<T>(GetAllBitsSetValue());
+            }
+        }
+
+        /// <summary>Constructs a vector whose components are all <paramref name="value" />.</summary>
+        [Intrinsic]
+        public unsafe Vector(T value) : this()
+        {
+            ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+
+            for (nint index = 0; index < Count; index++)
+            {
+                SetElement(index, value);
+            }
+        }
+
+        /// <summary>Constructs a vector from the given array. The size of the given array must be at least <see cref="Count" />.</summary>
+        [Intrinsic]
+        public unsafe Vector(T[] values) : this(values, 0) { }
+
+        /// <summary>
+        /// Constructs a vector from the given array, starting from the given index.
+        /// The array must contain at least <see cref="Count" /> from the given index.
+        /// </summary>
+        [Intrinsic]
+        public unsafe Vector(T[] values, int index) : this(values.AsSpan(index))
+        {
+            ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
+
+            if (values is null)
```

Comment:
Just a note: We currently enforce this to `throw` if `values` is null, which prevents us from being able to have it just forward to the `ROSpan` version.